Demo_project/models/event.go

```diff
@@ -0,0 +1,107 @@
+package models
+
+import (
+	"crud_project/db"
+	"time"
+	_ "github.com/pelletier/go-toml/query"
+)
+
+// 
+type Event struct{
```

Comment:
client side validation required 

You validate the DTO fields (e.g., required fields, data types, constraints). which should be in another file 
After validation, transform the DTO into a model or structure suitable for database storage.
Use the model to interact with the database (e.g., inserting, updating, querying data).

when the project complexity increase this could be right way to handle
example:
project/
├── dto/
│   ├── user_dto.go    // DTOs for user-related APIs
│   ├── product_dto.go // DTOs for product-related APIs
├── models/
│   ├── user_model.go    // Models for user database table
│   ├── product_model.go // Models for product database table
├── handlers/
│   ├── user_handler.go  // Handlers for user routes
│   ├── product_handler.go // Handlers for product routes

Reply:
and also add service folder where 
func CreateNewUser(name, email string) (*models.User, error) {
    user := &models.User{
        Name:  name,
        Email: email,
    }

    if err := database.CreateUser(user); err != nil {
        return nil, err
    }

    return user, nil
}

handler calls service functions - service functions calls - database function !

Reply:
service file sits in between handler and database - mainly handling business logic and also acts as a wrapper.
kindly go through this kind of medium/large project structure.